spring-ai-core/src/main/java/org/springframework/ai/vectorstore/VectorStore.java

```diff
@@ -30,107 +27,24 @@ default void accept(List<Document> documents) {
-	 * {@link Filter.Expression}.
-	 * @param topK the top 'k' similar results to return.
-	 * @param similarityThreshold the lower bound of the similarity score
-	 * @param filterExpression portable metadata filter expression.
-	 * @return similar documents that match the requested similarity threshold and filter.
+	 * exactly the number of nearest-neighbor results that match the request criteria.
+	 * @param request Search request for set search parameters, such as the query text,
+	 * topK, similarity threshold and metadata filter expressions.
+	 * @return Returns documents th match the query request conditions.
 	 */
-	default List<Document> similaritySearch(String query, int topK, double similarityThreshold,
-			Filter.Expression filterExpression) {
-		throw new UnsupportedOperationException("This vector store doesn't support search filtering");
-	}
+	List<Document> similaritySearch(SearchRequest request);
 
 	/**
-	 * Retrieves documents by query embedding similarity and metadata filters to retrieve
-	 * exactly the number of nearest-neighbor results that match the filters.
-	 *
-	 * For example if your {@link Document#getMetadata()} has a schema like:
-	 *
-	 * <pre>{@code
-	 * &#123;
-	 *  "country":  <Text>,
-	 *  "city":     <Text>,
-	 *  "year":     <Number>,
-	 *  "price":    <Decimal>,
-	 *  "isActive": <Boolean>
-	 * &#125;
-	 * }</pre>
-	 *
-	 * then you can constrain the search result with metadata filter expressions like:
-	 *
-	 * <pre>{@code
-	 *country == 'UK' && year >= 2020 && isActive == true
-	 *                        Or
-	 *country == 'BG' && (city NOT IN ['Sofia', 'Plovdiv'] || price < 134.34)
-	 * }</pre>
-	 *
-	 * This ensures that the response contains only embeddings that match the specified
-	 * filer criteria. <br/>
-	 *
-	 * The declarative, SQL like, filter syntax is portable across all vector stores
-	 * supporting the filter search feature.<br/>
-	 *
-	 * The {@link FilterExpressionTextParser} is used to convert the text filter
-	 * expression into {@link Filter.Expression}.
-	 * @param topK the top 'k' similar results to return.
-	 * @param similarityThreshold the lower bound of the similarity score
-	 * @param filterExpression portable metadata filter expression.
-	 * @return similar documents that match the requested similarity threshold and filter.
+	 * Retrieves documents by query embedding similarity using the default
+	 * {@link SearchRequest}'s' search criteria.
+	 * @param query Text to use for embedding similarity comparison.
+	 * @return Returns a list of documents that have embeddings similar to the query text
+	 * embedding.
 	 */
-	default List<Document> similaritySearch(String query, int topK, double similarityThreshold,
-			String filterExpression) {
-		var filterExpressionObject = Filter.parser().parse(filterExpression);
-		return similaritySearch(query, topK, similarityThreshold, filterExpressionObject);
+	default List<Document> similaritySearch(String query) {
```

Comment:
Just a thought, but we may need another overloaded metho that returns the List<Document>s but also some 'metadata' about the query.  I see that pinecone has the class QueryResponse which brings along additional metadata.    That could maybe go inside the document class itself under 'queryMetadata' or some other class that augments/wraps Document, DocumentWithQueryData ?